tensorflow/contrib/training/python/training/sgdr_learning_rate_decay_test.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Functional test for sgdr learning rate decay."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import math
+
+from sgdr_learning_rate_decay import sgdr_decay
+from tensorflow.python.platform import googletest
+from tensorflow.python.framework import test_util
+from tensorflow.python.ops import gen_state_ops
+from tensorflow.python.ops import state_ops
+from tensorflow.python.framework import dtypes
+
+
+class SGDRDecayTest(test_util.TensorFlowTestCase):
+
+  # Original iterative approach
+  # https://github.com/loshchil/SGDR/blob/master/SGDR_WRNs.py
+  def get_original_values(self, lr, Te, multFactor, iter_per_epoch, epochs):
+    t0 = math.pi / 2.0
+    tt = 0
+    TeNext = Te
```

Comment:
I tried to make as little changes compared to the original implementation as possible. I'll fix variable naming to match TensorFlow style.